### DISCUSSION TITLE
How to refresh cached user role

### DISCUSSION THREAD

**Discussion by ejkyecontentsx** [serenity-is/Serenity#6964] [Help]
Hi, 

When I create a user with a specific role then I add another role for the user,
In this case, the user can't access menus I assigned later. I guess the cached role hasn't refreshed.

Do you have any idea for this case?

**Comment by fatihozdemir23** [comment]
**I added the code I specified to the Create method in the NavigationModelFactory.cs file, and the problem was solved for me.**
public INavigationModel Create()
 {
**//add this code
     cache.Remove("LeftNavigationModel:NavigationItems:" + (userAccessor.User?.GetIdentifier() ?? "-1"));
//**    
 var model = new NavigationModel
     {
         Items = cache.GetLocalStoreOnly("LeftNavigationModel:NavigationItems:" +
             (userAccessor.User?.GetIdentifier() ?? "-1"),
             TimeSpan.Zero,
             UserPermissionRow.Fields.GenerationKey,
             () => NavigationHelper.GetNavigationItems(permissions, typeSource, serviceProvider, ToAbsolute))
     };

     CalcActivePath(model);

     return model;
 }

**Reply by ejkyecontentsx** [reply]
It works perfectly. Thanks!
